Added script createljspeech.py for easy dataset creation

thorstenMueller opened the pull request:
Sorry if this code is not optimized, but it's my first python script, so i tried my best.
It extracts information from mimic-recording-studio sqlite db table and creates a ljspeech-1.1 structure which can be used for futher processing.

krisgesling commented on createljspeech.py:
```diff
@@ -0,0 +1,54 @@
+# Directory of mimic-recording-studio (eg: /home/thorsten/mimic-recording-studio)
+directory_base_mrs = "/home/thorsten/tts/mimic-recording-studio"
+# Output directory for ljspeech structure (eg: /tmp/tts)
+directory_base_ljspeech = "/home/thorsten/tts"
```
Should this be set to the `tacotron` directory then users won't need to copy it across?

krisgesling replied:
Also could use [`os.path.expanduser`](https://stackoverflow.com/questions/4028904/how-to-get-the-home-directory-in-python#4028943) to generalize these paths

thorstenMueller replied:
Thanks for these hints. I will modify the script appropriate.

krisgesling commented on createljspeech.py:
```diff
@@ -0,0 +1,54 @@
+# Output directory for ljspeech structure (eg: /tmp/tts)
+directory_base_ljspeech = "/home/thorsten/tts"
+# See /mimic-recording-studio/backend/audio_files/<id> (eg: 3beeae88-0777-2c8c-5c93-2e844a462e26)
+speaker_id = "4aeeae88-0777-2c8c-5c93-2e844a462e49" 
```
can we detect this id from the file structure?

thorstenMueller replied:
I updated the script to query the userid from the sqlite db. But this just works if the db is just used by one user.
SELECT uuid FROM usermodel LIMIT 1;

In case there are more user entries in usermodel table i'd need to ask the user for its username to add this condition in the sql query.

krisgesling commented on README.md:
````diff
@@ -199,6 +206,12 @@ Example
     python analyze.py --train_file_path=~/tacotron/training/train.txt --save_to=~tacotron/visuals --cmu_dict_path=~/cmudict-0.7b
 ```
 
+In case of docker usage you've to export your DISPLAY variable on container start to successfully create the image files. Ensure x server connections are enabled on your host system.
````
Perhaps slightly rewording to:
> If using Docker, you need to export your DISPLAY variable when starting the container to successfully create the image files. Also ensure that X server connections are enabled on your host system.

thorstenMueller replied:
I will change the text to your proposal.

thorstenMueller commented:
Hey Kris.
I've updated the pull request on the basis of your feedback.

Now there's a mrs dataset property for native mimic-recording-studio data processing. I also moved/updated createljspeech.py to work without changing environment specific parameters withing the python script.

  ````
   python3 preprocess.py --dataset mrs --mrs_dir=<path_to>/mimic-recording-studio/
   ````

  ````
  python3 ./datasets/createljspeech.py --mrs_dir=<path_to>/mimic-recording-studio/
  ````

krisgesling commented on .gitignore:
```diff
@@ -3,4 +3,6 @@ __pycache__/
-.vscode
\ No newline at end of file
+.vscode
+export_recs_per_day.py
```
These seem to very specific to your machine, can we leave the `.gitignore` as is?

thorstenMueller replied:
You're right. This should not be part of merge request and .gitignore should stay unchanged.

thorstenMueller replied:
Commit rolled back

krisgesling commented on cpu.Dockerfile:
```diff
@@ -3,6 +3,8 @@ FROM tensorflow/tensorflow:1.8.0-py3
 COPY . /root/mimic2
 WORKDIR /root/mimic2
+RUN apt-get update
+RUN apt-get install -y python3-tk
```
Is this needed? TK is a GUI package right?

thorstenMueller replied:
Running it with original Dockerfile didn't work (for me). After installing/adding the python3-tk package it did work. But this should be verified or rolled back.

thorstenMueller replied:
Commit rolled back.
If it's needed i'll add a seperate pr in future.

thorstenMueller commented on datasets/createljspeech.py:
```diff
@@ -0,0 +1,47 @@
+
+    # Get user id from sqlite to find recordings in directory structure
+    # TODO: Currently just works with one user
+    for row in c.execute('SELECT uuid FROM usermodel LIMIT 1;'):
```
I thought about adding an optional parameter --username. If it's passed the correct guid is selected from the sqlite db. If it's not passed the first user from database is chosen (as it's currently). Then it would be possible to export the correct dataset, even if one mimic-recording-studio database is used by more than one reader.

@krisgesling What do you think on that.

thorstenMueller commented:
I rolled back the commits on Dockerfile and gitignore file and added support for multiple speakers in mimic-recording-studio using same sqlite database.

krisgesling commented on README.md:
````diff
@@ -199,6 +212,15 @@ Example
     python analyze.py --train_file_path=~/tacotron/training/train.txt --save_to=~tacotron/visuals --cmu_dict_path=~/cmudict-0.7b
 ```
 
+<<<<<<< HEAD
````
Looks like some artifacts got left behind from a merge conflict. If you delete these we should be good to go :)

Thanks heaps for all the work on this!

thorstenMueller replied:
Hopefully all unneeded artifacts should be removed now.

krisgesling commented:
Thanks for persevering and continuing to nudge me Thorsten :)

This is looking great. Merging now.
